### PR TITLE
feat(3d viewer): new prop highlightNodesOnComplete (#480)

### PR DESCRIPTION
**Pull request by nalaka**
Need to create mock data with asset-node mapping to test this.

### CODE REVIEW

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-538.gearboxjs.preview.cogniteapp.com

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-538.gearboxjs.preview.cogniteapp.com

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-538.gearboxjs.preview.cogniteapp.com

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-538.gearboxjs.preview.cogniteapp.com

**Comment by nalaka** [comment]
A new optional property `highlightNodesOnComplete: boolean` is introduced to control if `Model3DViewer` should highlight selected 3D nodes just before invoking the provided `onComplete` callback.

This defaults to true so as not to affect current behaviour without this property.

Unit test added to check if the component renders with this prop set to false.

File: src/components/Model3DViewer/Model3DViewer.tsx [hunk 1]
```diff
@@ -51,6 +51,7 @@ export interface Model3DViewerProps {
   boundingBox?: THREE.Box3;
   cache?: CacheObject;
   enableKeyboardNavigation?: boolean;
+  highlightNodesOnComplete?: boolean;
```
Comment (polomani) [inline src/components/Model3DViewer/Model3DViewer.tsx]:
```suggestion
  highlightNodes?: boolean;
```
but this basically means that nodes will never be highlighted?
so than maybe call it as just `highlightNodes`?

Comment (mrtraser) [reply src/components/Model3DViewer/Model3DViewer.tsx]: you should disable node highlighting in general (with this property). So better to move `highlightNodes` prop checking inside `highlightNodes` private method (you should change method name first)

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-538.gearboxjs.preview.cogniteapp.com

File: src/components/Model3DViewer/Model3DViewer.tsx (hunk 1 above)
Comment (nalaka) [reply src/components/Model3DViewer/Model3DViewer.tsx]: Implemented these suggestions - but kept the private method name as it doesn't conflict with the same named prop name

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-538.gearboxjs.preview.cogniteapp.com

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-538.gearboxjs.preview.cogniteapp.com

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-538.gearboxjs.preview.cogniteapp.com

File: src/components/Model3DViewer/Model3DViewer.tsx [hunk 2]
```diff
@@ -516,6 +518,11 @@ export class Model3DViewer extends React.Component<Model3DViewerProps> {
   }
 
   private highlightNodes() {
+    const { highlightNodes } = this.props;
+    if (!highlightNodes) {
+      return;
```
Comment (polomani) [inline src/components/Model3DViewer/Model3DViewer.tsx]: more up to @mrtraser but I believe that it's not very according to a clean code that this function doesn't always highlight nodes. So for me you should either put that logic `if (highlightNodes) { call highlightNodes () }` outside or rename the function to something like `highlightNodesIfAllowed`

File: src/components/Model3DViewer/stories/full.md [hunk 3]
```diff
@@ -81,6 +81,7 @@ function ExampleComponent(props) {
 | `assetId`                  | id of asset to highlight                                                                             | `number`                                                                            |         |
 | `boundingBox`              | bounding box object, that describes dimension of viewed asset nodes                                  | `THREE.Box3`                                                                        |         |
 | `cache`                    | object for caching 3D viewers instances                                                              | `{ [name:string]: any }`                                                            |         |
+| `highlightNodes`           | enable node highlighting behaviour                                                                    | `boolean`                                                                           | true    |
```
Comment (polomani) [inline src/components/Model3DViewer/stories/full.md]: not very clear what it does. if I never seen it I wouldn't understand, does it highlights all nodes of the asset?

Comment (nalaka) [reply src/components/Model3DViewer/stories/full.md]: It highlights all nodes that are mapped to some asset - @mrtraser please correct if I am wrong?

@polomani - Maybe we should name the property `highlightMappedNodes` to make the intent obvious?

File: src/components/Model3DViewer/Model3DViewer.tsx (hunk 2 above)
Comment (nalaka) [reply src/components/Model3DViewer/Model3DViewer.tsx]: @mrtraser - what do you think?

Comment (mrtraser) [reply src/components/Model3DViewer/Model3DViewer.tsx]: As I mentioned above, you definitely need to change method name.  About naming – `highlightMappedNodes` is ok for me.

File: src/components/Model3DViewer/stories/full.md (hunk 3 above)
Comment (mrtraser) [reply src/components/Model3DViewer/stories/full.md]: `enable node highlighting behaviour` -> `enable default highlighting logic for mapped to provided assetId nodes`

File: src/components/Model3DViewer/Model3DViewer.tsx (hunk 2 above)
Comment (nalaka) [reply src/components/Model3DViewer/Model3DViewer.tsx]: Changed:

- Property name to `highlightMappedNodes`
- Private method name to `highlightMappedNodesIfAllowed `
- Updated property description to `enable default highlighting of 3D nodes that are associated with some asset`

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-538.gearboxjs.preview.cogniteapp.com

File: src/components/Model3DViewer/stories/full.md (hunk 3 above)
Comment (nalaka) [reply src/components/Model3DViewer/stories/full.md]: Updated the property description to `enable default highlighting of 3D nodes that are associated with some asset`

**Comment by nalaka** [comment]
@mrtraser - I think this is now ready, please review and let me know if it is ok to merge.

**Pull request merged**